tools/nx-plugins/npm-package/build/index.ts

```diff
@@ -46,20 +46,23 @@ const getRollupConfig = (
 		compilerOptions.target = ScriptTarget.ES2018;
 	}
 
+	/** only compile TypeScript sources */
```

Comment:
how do we guarantee type safety at build without this?

Reply:
also, don't we still want to generate declaration files?

Reply:
I did not realise we did not have a separate step for that.

Maybe using [`noEmit`](https://www.typescriptlang.org/tsconfig#noEmit)?

Reply:
That seems to have done the trick, as I managed to get a build error:

<img width="885" alt="image" src="https://user-images.githubusercontent.com/76776/231499134-3eabc69a-c4df-4393-a683-4b69ccdbc89a.png">